src/visualization/recipes_plots.jl

```diff
@@ -5,20 +5,45 @@ const TrixiParticlesODESolution = ODESolution{<:Any, <:Any, <:Any, <:Any, <:Any,
                                                            <:Semidiscretization}}
 
 RecipesBase.@recipe function f(sol::TrixiParticlesODESolution)
-    # Redirect everything to the recipe
+    # Redirect everything to the next recipe
     return sol.u[end].x..., sol.prob.p
 end
 
+# GPU version
+RecipesBase.@recipe function f(sol::TrixiParticlesODESolution, semi::Semidiscretization)
+    # Move GPU data to the CPU
+    v_ode = Array(sol.u[end].x[1])
+    u_ode = Array(sol.u[end].x[2])
+    semi_ = Adapt.adapt(Array, sol.prob.p)
+
+    # Redirect everything to the next recipe
+    return v_ode, u_ode, semi_, particle_spacings(semi)
+end
+
+RecipesBase.@recipe function f(v_ode::AbstractGPUArray, u_ode, semi::Semidiscretization;
+                               particle_spacings=nothing,
+                               size=(600, 400), # Default size
+                               xlims=(-Inf, Inf), ylims=(-Inf, Inf))
+    throw(ArgumentError("to plot GPU data, use `plot(sol, semi)`"))
+end
+
 RecipesBase.@recipe function f(v_ode, u_ode, semi::Semidiscretization;
+                               particle_spacings=TrixiParticles.particle_spacings(semi),
```

Comment:
```suggestion
                               particle_spacings=particle_spacings(semi),
```
Why do you need this? It's defined in the module scope?